TransactionService/docker_context/handlers/LegacyStockServerHandler.py

```diff
@@ -0,0 +1,60 @@
+import socket
+from time import sleep
+from datetime import datetime
+import pickle
+
+
+class LegacyStockServerHandler:
+
+    def __init__(self, ip, port, redis, audit):
+        self.ip = ip
+        self.port = port
+        self.redis = redis
+        self.audit = audit
+
+    def quoteSocket(self):
+        stockSocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        print("Attempting Connection to Quote Server")
+        while stockSocket.connect_ex((self.ip, self.port)) != 0:
+            sleep(1)

```

Comment:
could this potentially hang execution of the service, therefore, the user if we were unable to connect? could implement a timeout for it

Reply:
You are right that this could hang the operation, but only if the legacy stock server is down, in which case there are likely bigger issues. I can include a more proper timeout, but it would likely just end up forcing an error back through to the web server saying that we couldn't establish a connection to the quote server.